### PR TITLE
Update button styles

### LINKED ISSUE #13: Button component
Something along these lines:

Screenshot 2019-05-19 at 00 50 54

Needs:

- [ ] Basic Sketch components
- [ ] React components
- [ ] Documentation

### PR DESCRIPTION
**Pull request by sarahmonster**
Closes #13.

This is my first pass at button styles. There's a whoooooole lot of messy repeatedness in the code we could probably clean up, but I did my best with what I could. You can see the disabled button tries to apply styles (basically just "convert anything that is pink to grey" which is also what the intent [green/yellow/red] buttons will do) but doesn't quite make it. 

Screenshot 2019-05-19 at 03 48 44

It would be super groovy if we could just say something like:

```
primary {
  border: $currentButtonColour;
  background: $currentButtonColour;
  color: white;
}

minimal {
  border-bottom: 2px solid $currentButtonColour;
  colour: $currentButtonColour;
}
```

where $currentButtonColour equals the colour defined by `intent` or a `disabled` state. (Actually, maybe disabled belongs with intent, rather than state, since these are where we're using colour?) We'd also need a $currentButtonColourLight and $currentButtonColourDark to add shadows and highlights as needed, but I figure if we can get one variable we could probably also get the other...

Screenshot 2019-05-19 at 00 50 54

You can see there are defined patterns for the buttons, and the only thing that changes across variations is the colour used. 

```
  primary: no border, colour background, white text
  default: colour border, white background, colour text
  minimal: bottom border, no background, colour text
  hover: box-shadow dark colour
  focus: box-shadow light colour
  active: background dark colour
```

This would probably also make themeing easier.

This is what I originally had for button colour variables:

```
export const buttonColors = {
  // colour corresponds to state or intent
  default: colors.blue,
  hover: colors.darkBlue,
  active: colors.lightBlue,
  disabled: colors.grey,
  danger: colors.red,
  success: colors.green,
  warning: colors.yellow,
  primaryText: colors.white,
};
```

### CODE REVIEW

**Comment by sarahmonster** [comment]
Updated this! There are now semantic colours for each different intent/state that uses it, and it's entirely relying on theme colours. 🎊

I've also fleshed out the documentation a wee bit, although it could use some diagrams and more detail later. I've added a tracking issue (#5) to keep track of potential future-improvements as well, but I think this should work for our 0.1.0 purposes.

**Comment by sarahmonster** [comment]
Oh, I updated the active styling here ... I think it should be a lot more usable. There may be some accessibility contrast issues, but we'll handle those once a) my`tota11y` extension is working again and b) colour variables are stable, which they are currently not. (Ideally the numbers would ensure contrast and the yellow will be a lot less pukey-looking).

I tried to underline it in the documentation, but the idea is that you'd only ever use one primary button per screen—they should be used for the Most Important Action on that page.

Disabled buttons should be better now too.

I'm not entirely in ❤️ with the hover styling, but I'm going to revisit that again later.

File: src/components/Button/README.md [hunk 1]
```diff
@@ -1,75 +1,146 @@
-Use a button any time the user should complete an action.
+A button indicates that a user can complete an action or make a choice, using a single click or tap. They are used for submitting forms, navigating to new pages, as well as toolbars and utilities.
 
 ```js
 const onClick = (event) => {
-  window.alert('🚨 Alarm sounded! 🚨');
+	window.alert('🚨 Alarm sounded! 🚨');
```
Comment (tofumatt) [inline src/components/Button/README.md]: 😮 TABS!

I'll need to get some kind of lint setup for this. It's fine for now.

File: src/components/Button/index.js [hunk 2]
```diff
@@ -3,67 +3,135 @@ import invariant from 'invariant';
...
 import React from 'react';
 
+import { useTheme } from 'themes';
+
 export const Button = ({
   children,
   minimal,
   navigation,
   primary,
+  disabled,
```
Comment (tofumatt) [inline src/components/Button/index.js]: I'd like these to be alphabetised, but again: I'll find a linter that'll handle it.

**Pull request merged**